src/kvstore/RocksEngine.cpp

```diff
@@ -217,6 +219,7 @@ ResultCode RocksEngine::rangeWithPrefix(const std::string& start,
                                         const std::string& prefix,
                                         std::unique_ptr<KVIterator>* storageIter) {
     rocksdb::ReadOptions options;
+    options.prefix_same_as_start = true;
```

Comment:
A question，if set options.prefix_same_as_start = true in this method, Is the check as below redundant of `RocksPrefixIter`?

```
bool valid() const override {
        return !!iter_ && iter_->Valid() && (iter_->key().starts_with(prefix_));
    }
```

Reply:
I think so, let me dig it more

Reply:
Seems prefix_same_as_start can save one Next operation, so we’d better keeping it here?

Reply:
> Seems prefix_same_as_start can save one Next operation, so we’d better keeping it here?

I agree，good that.

Reply:
Can ScanProcessor work well with prefix_same_as_start on.

Reply:
Hi, ScanProcessor will use RocksEngine::range instead of RocksEngine::rangeWithPrefix？